Add page.on('metric') docs (#1807)
grafana/k6-docs: docs/sources/k6/next/javascript-api/k6-browser/page/on.md

| event     | string   | `''`    | Event to attach the handler to. Currently, `'console'` and `'metric'` events are supported. |
| handler   | function | `null`  | A function to be called every time the specified event is emitted.                          |

### Events

| Event     | Description                                                                                                                                                                                                                                                          |
| --------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `console` | Emitted every time the console API methods are called from within the page JavaScript context. The arguments passed into the handler are defined by the [`ConsoleMessage`](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-browser/consolemessage) class. |

> | `metric`  | Emitted every time a metric is measured and emitted for the page. The arguments passed into the handler are defined by the [`MetricMessage`](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-browser/metricmessage) class.                                |

Reviewer: To avoid confusion, I would rename "class" with "object" as we say a `MetricMessage` object in the introduction.

Author: Yep, makes sense, updated in f45cda8df1d61df4e201f0a01f6ec8194916ed27

```

{{< /code >}}

### Metric event Example

{{< code >}}

```javascript
import { browser } from 'k6/browser';

export const options = {
  scenarios: {
    ui: {
      executor: 'shared-iterations',
      options: {
        browser: {
          type: 'chromium',
        },
      },
    },
  },
};

export default async function () {
  const page = await browser.newPage();

>   // Here we call page.on('metric). It's possible to call page.on('metric') more
>   // than once, and the callback function will be executed in the order page.on
>   // was called.

Reviewer: I'd explain what we're doing first. And then explain the other possible use cases for clarity.

```suggestion
  // We first register a handler using page.on('metric').
  // Calling page.on('metric') multiple times are allowed. The registered handlers
  // will be executed in the order page.on was called.
```

Author: Thanks for the suggestion! I went with:

```suggestion
  // We first register a handler using page.on('metric'). Calling page.on('metric')
  // multiple times is allowed. The registered handlers will be executed in the
  // order page.on was called.
```

f27607252bf72393d0a06e12f74220e71a2dba0e

  page.on('metric', (metric) => {

>     // At the moment metric.tag is the only method on the metricMessage object.
>     // It will find a match between the current metric url and name tags against
>     // the supplied regular expression in `url`.

Reviewer: Same here, usage first:

```suggestion
    // Using metric.tag finds a match between the current metric url and name
    // tags against the supplied regular expression in `url`.
    //
    // At the moment metric.tag is the only method on the metricMessage object.
```
